dev/run-tests.py

```diff
@@ -176,8 +176,9 @@ def determine_java_version(java_exe):
     # find raw version string, eg 'java version "1.8.0_25"'
     raw_version_str = next(x for x in raw_output_lines if " version " in x)
 
-    version_str = raw_version_str.split()[-1].strip('"')  # eg '1.8.0_25'
-    version, update = version_str.split('_')  # eg ['1.8.0', '25']
+    version_str = raw_version_str.split()[-1].strip('"')  # eg '1.8.0_25-internal'
```

Comment:
Shall we take this a step further and only parse out something like `\d(\.\d+){2,}(_\d+)?` to be more explicit about exactly what to accept?